frontend/packages/ux-editor/src/components/config/componentSpecificContent/AttachmentList/AttachmentListInternalFormat.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import { Switch } from '@digdir/design-system-react';
+import { AttachmentListContent } from './AttachmentListContent';
+import { useTranslation } from 'react-i18next';
+import { ArrayUtils } from '@studio/pure-functions';
+import { currentTasks, getAvailableAttachments, reservedDataTypes } from './AttachmentListUtils';
+import type { LayoutSets } from 'app-shared/types/api/LayoutSetsResponse';
+import type { ApplicationMetadata, DataTypeElement } from 'app-shared/types/ApplicationMetadata';
+
+type AttachmentListInternalFormatProps = {
+  handleOutGoingData: (selectedDataTypes: string[], availableAttachments: string[]) => void;
+  internalDataFormat: {
+    availableAttachments: string[];
+    selectedDataTypes: string[];
+  };
+  layoutSets: LayoutSets;
+  selectedLayoutSet: string;
+  appMetadata: ApplicationMetadata;
+};
+
+export const AttachmentListInternalFormat = (props: AttachmentListInternalFormatProps) => {
+  const { handleOutGoingData, internalDataFormat, layoutSets, selectedLayoutSet, appMetadata } =
+    props;
+  const { availableAttachments } = internalDataFormat;
+  const { t } = useTranslation();
+  const [selectedDataTypes, setSelectedDataTypes] = useState<string[]>(
+    internalDataFormat.selectedDataTypes,
+  );
+  const includePdf = selectedDataTypes.includes(reservedDataTypes.refDataAsPdf);
+  const currentTask = selectedDataTypes.includes(reservedDataTypes.currentTask);
+
+  const onChangePdf = (isChecked: boolean) => {
+    const updatedSelection = toggleItemInArray(
+      selectedDataTypes,
+      reservedDataTypes.refDataAsPdf,
+      isChecked,
+    );
+
+    setSelectedDataTypes(updatedSelection);
+    handleOutGoingData(updatedSelection, availableAttachments);
+  };
+
+  const onChangeTask = (isChecked: boolean) => {
+    let updatedSelection: string[];
+
+    if (isChecked) {
+      updatedSelection = getAvailableSelectedDataTypes(
+        layoutSets,
+        selectedLayoutSet,
+        appMetadata.dataTypes,
+        selectedDataTypes,
+        availableAttachments,
+        includePdf,
+      );
+    } else {
+      updatedSelection = toggleItemInArray(
+        selectedDataTypes,
+        reservedDataTypes.currentTask,
+        isChecked,
+      );
+    }
```

Comment:
Hvorfor er det to helt forksjellige funksjoner som kalles ut fra om sjekkboksen er avkrysset eller ikke?